minions/async/src/main/java/io/github/microcks/minion/async/consumer/NATSMessageConsumptionTask.java

```diff
@@ -0,0 +1,175 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An implementation of <code>MessageConsumptionTask</code> that consumes a
+ * topic on an NATS. Endpoint URL should be specified using the
+ * following form:
+ * <code>nats://{brokerhost[:port]}</code>
+ *
+ * @author laurent
+ */
+public class NATSMessageConsumptionTask implements MessageConsumptionTask {
+
+    /**
+     * Get a JBoss logging logger.
+     */
+    private final Logger logger = Logger.getLogger(getClass());
+
+    /**
+     * The string for Regular Expression that helps validating acceptable
+     * endpoints.
+     */
+    public static final String ENDPOINT_PATTERN_STRING = "nats://(?<brokerUrl>[^:]+(:\\d+)?)";
+    /**
+     * The Pattern for matching groups within the endpoint regular expression.
+     */
+    public static final Pattern ENDPOINT_PATTERN = Pattern.compile(ENDPOINT_PATTERN_STRING);
+
+    private File trustStore;
+
+    private AsyncTestSpecification specification;
+
+    private Connection subscriber;
+
+    private String endpointTopic;
+
+    private String options;
+
+    /**
+     * Create a new consumption task from an Async test specification.
+     *
+     * @param testSpecification The specification holding endpointURL and
+     * timeout.
+     */
+    public NATSMessageConsumptionTask(AsyncTestSpecification testSpecification) {
+        this.specification = testSpecification;
+    }
+
+    /**
+     * Convenient static method for checking if this implementation will accept
+     * endpoint.
+     *
+     * @param endpointUrl The endpoint URL to validate
+     * @return True if endpointUrl can be used for connecting and consuming on
+     * endpoint
+     */
+    public static boolean acceptEndpoint(String endpointUrl) {
+        return endpointUrl != null && endpointUrl.matches(ENDPOINT_PATTERN_STRING);
+    }
+
+    @Override
+    public List<ConsumedMessage> call() throws Exception {
+        if (subscriber == null) {
+            intializeNATSClient();
+        }
+        List<ConsumedMessage> messages = new ArrayList<>();
+
+        // Start subscribing to the broker endpoint topic.
+        Dispatcher d = subscriber.createDispatcher((msg) -> {
+            String msgString = new String(msg.getData(), StandardCharsets.UTF_8);
+            logger.info("Received a new NATS Message: " + msgString);
+            // Build a ConsumedMessage from NATS message.
+            ConsumedMessage message = new ConsumedMessage();
+            message.setReceivedAt(System.currentTimeMillis());
+            message.setHeaders(buildHeaders(msg.getHeaders()));
+            message.setPayload(msg.getData());
+            messages.add(message);
+        });
+
+        d.subscribe("subject");
```

Comment:
I think it should be replaced with a subject name that would be acquired from parsing the endpoint url.

Reply:
Yep 👍